### ISSUE TITLE
No THC/THC.h when building horovod 0.23.0 with pytorch 1.11.0-rc4 with CUDA 11.5

### ISSUE THREAD

**Issue by ydaiming** [horovod/horovod#3425]
When trying to install horovod 0.23.0 together with pytorch 1.11.0-rc4, NCCL 2.11.4, and CUDA 11.5, the following error showed up near the end of building:
```
      /tmp/pip-install-3xg76d_2/horovod_02cea17fbf3747bb8ce9d6b7c589b182/horovod/torch/cuda_util.cc:18:10: fatal error: THC/THC.h: No such file or directory
         18 | #include <THC/THC.h>
            |          ^~~~~~~~~~~
      compilation terminated.
      make[2]: *** [horovod/torch/CMakeFiles/pytorch.dir/build.make:576: horovod/torch/CMakeFiles/pytorch.dir/cuda_util.cc.o] Error 1
      make[2]: Leaving directory '/tmp/pip-install-3xg76d_2/horovod_02cea17fbf3747bb8ce9d6b7c589b182/build/temp.linux-x86_64-3.8/RelWithDebInfo'
      make[1]: *** [CMakeFiles/Makefile2:486: horovod/torch/CMakeFiles/pytorch.dir/all] Error 2
      make[1]: Leaving directory '/tmp/pip-install-3xg76d_2/horovod_02cea17fbf3747bb8ce9d6b7c589b182/build/temp.linux-x86_64-3.8/RelWithDebInfo'
      make: *** [Makefile:149: all] Error 2
```

The following commands were used to install `horovod`:
```
RUN pip uninstall -y horovod \
 && ldconfig /usr/local/cuda-11.5/targets/x86_64-linux/lib/stubs \
 && HOROVOD_GPU_ALLREDUCE=NCCL HOROVOD_CUDA_HOME=/usr/local/cuda-11.5 HOROVOD_WITH_PYTORCH=1 pip install --no-cache-dir horovod==${HOROVOD_VERSION} \
 && ldconfig
```

Does `horovod` support CUDA 11.5 yet?

**Comment by ydaiming** [comment]
Hi, horovod team. I think the latest version doesn't have this issue now.
https://github.com/horovod/horovod/blob/642a6b3018a84a0342c088165031f9b74e92f188/horovod/torch/cuda_util.cc
When do we expect a new version release for the `horovod` to accommodate the PyTorch 1.11.0 release?

**Comment by maxhgerlach** [comment]
Hi @ydaiming,
this build error with recent PyTorch has been fixed in Horovod 0.24.0, released today. It shouldn't be directly related to your version of CUDA.
Would you close the issue if your problem is resolved with the new stable version of Horovod?

**Comment by ydaiming** [comment]
Thank you @maxhgerlach , the new stable version solves the issue.
